### PR TITLE
`Dropdown` - Focus ring on click for `ToggleIcon` fix

### PR DESCRIPTION
**Pull request by dchyun**
### :pushpin: Summary

If merged, this PR would fix an open issue with the `ToggleIcon` variation of the `Dropdown` where the focus ring does not appear on mouse click on the button. 

### 🛠️ Detailed description

Currently the `ToggleIcon` variation of the `Dropdown` only shows a focus ring when the menu is opened, if it is opened via the keyboard, not the mouse. This interaction differs from that of the `ToggleButton` variation of the `Dropdown` and from the generic `Button` component. 

The approach for this variation has been update to align with those of other button components in the library. It now will show the focus ring on both mouse click and keyboard interaction. 

### :camera_flash: Screenshots

Current behavior on mouse click
Screenshot 2024-11-20 at 9 09 42 AM

New behavior on mouse click
Screenshot 2024-11-20 at 9 09 35 AM

### :link: External links

Jira ticket: [HDS-4096](https://hashicorp.atlassian.net/browse/HDS-4096)

***

### 👀 Component checklist

- [x] Percy was checked for any visual regression
- [x] A changelog entry was added via [Changesets](https://github.com/changesets/changesets) if needed (see [templates here](https://hashicorp.atlassian.net/wiki/spaces/HDS/pages/3243114706/Changelog+authoring+best+practices#Templates))

:speech_balloon: Please consider using [conventional comments](https://conventionalcomments.org/) when reviewing this PR.

[HDS-4096]: https://hashicorp.atlassian.net/browse/HDS-4096?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by alex-ju — COMMENTED** [review]
The active (+ focus) state should ideally match the one we have for Dropdown Toggle (and Button); which is essentially the active state.

dropdown-focus

File: packages/components/src/styles/components/dropdown.scss [hunk 1]
```diff
@@ -28,6 +28,7 @@ $hds-dropdown-toggle-border-radius: $hds-button-border-radius;
 // TOGGLE/ICON
 
 .hds-dropdown-toggle-icon {
+  position: relative;
```
Comment (alex-ju) [inline packages/components/src/styles/components/dropdown.scss:31]: noting here that this element does not contain any custom content (we control all the elements nested within) to there is not risk of introducing layout/positioning issue to our consumers.

**Review by alex-ju — APPROVED** [review]
Works as expected and the CSS implementation seems in line with the Button component! Let's get some design eyes on it and then we can get it in.

**Review by zamoore — APPROVED** [review]
Nice work

**Pull request merged**